canonical-web-and-design/maas-ui
# Improve search DSL and add selected filtering

## Done
- Clean up DSL destructuring.
- Implement selected filtering.
- Fix DSL bugs.

## QA
- Visit /r/machines.
- Select some machines.
- In the search box search for `in:selected`, you should only see your selected machines.
- In the search box search for `in:!selected`, you should not see your selected machines.
- Check that you can search in other ways according to: https://maas.io/docs/interactive-search

## Fixes
Fixes: https://github.com/canonical-web-and-design/maas-squad/issues/1799.
